package/network/utils/comgt/files/ncm.sh

```diff
@@ -71,10 +71,25 @@ proto_ncm_setup() {
 		return 1
 	}
 
-	[ -n "$delay" ] && sleep "$delay"
-
-	manufacturer=$(gcom -d "$device" -s /etc/gcom/getcardinfo.gcom | awk 'NF && $0 !~ /AT\+CGMI/ { sub(/\+CGMI: /,""); print tolower($1); exit; }')
-	[ $? -ne 0 -o -z "$manufacturer" ] && {
+	start=$(date +%s)
+	while true; do
+		manufacturer=$(gcom -d "$device" -s /etc/gcom/getcardinfo.gcom | awk 'NF && $0 !~ /AT\+CGMI/ { sub(/\+CGMI: /,""); print tolower($1); exit; }')
+		[ "$manufacturer" = "error" ] && {
```

Comment:
Maybe this could be simplified by only checking if the returned `manufacturer` is part of the supported manufacturer list.
If it is not supported -> **retry**.
If the manufacturer is supported -> **break**

This also catches errors if a wrong return value (not only `error`) has been received.